### PR TITLE
Add audio submission comment UI

### PR DESCRIPTION
**Pull request by vanwagonet**
This does not yet include uploading and submitting the comment.
extra: aligns use of MediaCommentType
extra: reusable keyboard avoiding
extra: consistency in empty states for drawer panels

refs: MBL-11318
affects: none
release note: none

### CODE REVIEW

**Comment by inst-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  This pull request will not generate a release note.
  </td>
    </tr>
  </tbody>
</table>


#### Affected Apps: None

[MBL-11318](https://instructure.atlassian.net/browse/MBL-11318)

Coverage | New % | Master % | Delta
-------- | ----- | -------- | -----
**Core** | 93.78% | 93.78% | 0.01%
**Student** | 93.16% | 93.86% | -0.7%
**Teacher** | 100% | 100% | 0%
**React Native** | 92.54% | 92.54% | 0%
<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: Core/Core/Views/KeyboardTransitioning.swift [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+//
+
+import UIKit
+
+/// To successfully avoid the keyboard, you need to animate a constraint in sync with the keyboard.
```
Comment (armstrongnate) [inline Core/Core/Views/KeyboardTransitioning.swift]: What is the behavior without this?

File: Core/Core/Views/KeyboardTransitioning.swift [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+/// Initialize a `KeyboardTransitioning` in `viewDidAppear` and save it as a property.
+public class KeyboardTransitioning {
```
Comment (armstrongnate) [inline Core/Core/Views/KeyboardTransitioning.swift]: Thanks for abstracting this.

**Comment by armstrongnate** [comment]
UI looks great.

File: Core/Core/Views/KeyboardTransitioning.swift (hunk 1 above)
Comment (vanwagonet) [reply Core/Core/Views/KeyboardTransitioning.swift]: The keyboard just covers your UI, and you probably can't ever get to what's under it.

**Pull request merged**